src/record.rs

```diff
@@ -0,0 +1,193 @@
+    ///         0x3B,0x2A,0x02,0x00,0x00,0x00,0x07,0x00,0x00,0x00,0x80,0xBC,0x04,0x00,0x00,0x00,
+    ///         0x53,0xC7,0x8B,0x18,0xC5,0xCC,0xCE,0x01,0x02,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+    ///         0x00,0x00,0x00,0x00,0x20,0x20,0x00,0x00,0x20,0x00,0x3C,0x00,0x42,0x00,0x54,0x00,
+    ///         0x44,0x00,0x65,0x00,0x76,0x00,0x4D,0x00,0x61,0x00,0x6E,0x00,0x61,0x00,0x67,0x00,
+    ///         0x65,0x00,0x72,0x00,0x2E,0x00,0x6C,0x00,0x6F,0x00,0x67,0x00,0x00,0x00,0x00,0x00
+    ///     ];
+    /// 
+    ///     let record = match record::UsnRecordV2::new(&mut Cursor::new(record_buffer)) {
+    ///         Ok(record) => record,
+    ///         Err(error) => panic!(error)
+    ///     };
+    /// 
+    ///     assert_eq!(record.record_length, 96);
+    ///     assert_eq!(record.major_version, 2);
+    ///     assert_eq!(record.minor_version, 0);
+    ///     assert_eq!(record.file_reference.entry, 115);
+    ///     assert_eq!(record.file_reference.sequence, 37224);
+    ///     assert_eq!(record.parent_reference.entry, 141883);
+    ///     assert_eq!(record.parent_reference.sequence, 7);
+    ///     assert_eq!(record.usn, 20342374400);
+    ///     assert_eq!(record.timestamp.0, 130266586132760403);
+    ///     assert_eq!(format!("{}", record.timestamp), "2013-10-19 12:16:53.276040");
+    ///     assert_eq!(record.reason.bits(), 2);
+    ///     assert_eq!(record.source_info.bits(), 0);
+    ///     assert_eq!(record.security_id, 0);
+    ///     assert_eq!(record.file_attributes, 8224);
+    ///     assert_eq!(record.file_name_length, 32);
+    ///     assert_eq!(record.file_name_offset, 60);
+    ///     assert_eq!(record.file_name, "BTDevManager.log");
+    /// # }
+    /// ```
+    pub fn new<T: Read>(mut buffer: T) -> Result<UsnRecordV2, UsnError> {
+        let record_length = buffer.read_u32::<LittleEndian>()?;
+
+        // Do some length checks
+        if record_length == 0 {
+            return Err(
+                UsnError::invalid_v2_record(
+                    format!("Record length is 0.")
+                )
+            );
+        }
+        if record_length > 1024 {
+            return Err(
+                UsnError::invalid_v2_record(
+                    format!("Record length is over 1024.")
+                )
+            );
+        }
+
+        let major_version = buffer.read_u16::<LittleEndian>()?;
+        if major_version != 2 {
+            return Err(
+                UsnError::invalid_v2_record(
+                    format!("Major version is not 2")
+                )
+            );
+        }
+
+        let minor_version = buffer.read_u16::<LittleEndian>()?;
+        if minor_version != 0 {
+            return Err(
+                UsnError::invalid_v2_record(
+                    format!("Minor version is not 0")
+                )
+            );
+        }
+
+        let file_reference = MftReference::from_reader(&mut buffer).unwrap();
```

Comment:
Don't unwrap here, wrap the error from `winstructs` with your own error instead :) (implement From<>..)